Quadriga/src/main/java/edu/asu/spring/quadriga/service/transformation/impl/TransformationManager.java

```diff
@@ -51,4 +55,80 @@ public void saveTransformations(ITransformationFile transformationFile) throws F
         return transformationDAO.getAllTransformations();
     }
 
+    /**
+     * Retrieve the transformation file object which contains the absolute file
+     * paths of patternFile and MappingFile .
+     * 
+     * @param transformationId
+     *            Transformation ID of the transformation.
+     * @return Returns the transformation file object
+     */
+    @Transactional
+    @Override
+    public ITransformationFile getTransformationFile(String transformationId) {
+
+        TransformFilesDTO transformDTO = (TransformFilesDTO) transformationDAO.getDTO(transformationId);
+
+        String fileLocation = transformationFileService.getTransformFileLocation();
+        String absolutePatternFilePath = getAbsoluteFilePath(fileLocation, transformationId,
+                transformDTO.getPatternFileName());
+        String absoluteMappingFilePath = getAbsoluteFilePath(fileLocation, transformationId,
+                transformDTO.getMappingFileName());
+
+        TransformationFile transformFile = new TransformationFile();
+        transformFile.setAbsolutePatternFilePath(absolutePatternFilePath);
+        transformFile.setAbsoluteMappingFilePath(absoluteMappingFilePath);
+        transformFile.setTitle(transformDTO.getTitle());
+        transformFile.setDescription(transformDTO.getDescription());
+        transformFile.setPatternFileName(transformDTO.getPatternFileName());
+        transformFile.setPatternDescription(transformDTO.getMappingDescription());
+        transformFile.setPatternTitle(transformDTO.getPatternTitle());
+        transformFile.setMappingFileName(transformDTO.getMappingFileName());
+        transformFile.setMappingDescription(transformDTO.getMappingDescription());
+        transformFile.setMappingTitle(transformDTO.getMappingTitle());
+        transformFile.setUserName(transformDTO.getUserName());
```

Comment:
this should be done in a mapper class. take a look at the ProjectHandleResolverMapper.